id_infrastructure/firestore_uuid_table.py

```diff
@@ -21,6 +21,43 @@ def __init__(self, table_name, crypto_token_path, uuid_prefix):
         self._table_name = table_name
         self._uuid_prefix = uuid_prefix
 
+    def set_mappings(self, mappings):
+        """
+        Sets uuids for data items using the given table of mappings from data to uuids.
+
+        :param mappings: Mappings from data to uuids
+        :type mappings: dict of str -> str
+        """
+        print(f"Setting {len(mappings)} mappings...")
+
+        # Ensure that the requested uuids are in the correct format for this table
+        for uuid in mappings.values():
+            assert uuid.startswith(self._uuid_prefix), f"UUID {uuid} does not start with the uuid prefix {self._uuid_prefix}"
+
+        # Batch write the mappings
```

Comment:
We need to ensure that this isn't trampling on any existing data.

- Batch read the table
- Verify that all the mappings we are trying to write are consistent i.e. if X : A in the old dataset, and we are being asked to set X : Y, then crash before doing any writes
- Drop all the writes that already exist in the table to decrease write costs.

Reply:
changed to do this